packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -694,6 +696,8 @@ class TimeInput extends React.Component {
         }
       }
       event.preventDefault();
+    } else if (event.keyCode === KeyCode.KEY_RIGHT) {
+      this.setState({ meridiemFocused: false });
```

Comment:
This seems to have fixed the issue with the right arrow key changing the meridiem value.
But I noticed that after the right arrow key is pressed and the meridiem field no longer has focus, when the left arrow key is pressed, it skips the meridiem field and focuses the seconds/minutes field. This to me felt counter-intuitive, since I was expecting it to go back to the meridiem field, i.e. Left arrow key navigation sequence to be same as right arrow key navigation sequence.

Reply:
@avinashg1994 @supreethmr I agree that the right arrow key can change the meridiem value seems counter-intuitive. It makes to add logic to prevent that from happening. It does appear that the changes made above to set the meridiemFocused state to false when the key is KEY_RIGHT causes the focus to lose as @avinashg1994 mentioned. One option is instead of relying on using the meridiemFocused state to handle this, you could create a new variable (i.e. preventMeridiemChange) and set it to true on line 700. Then in handleMeridiemChange, you can skip the setting the meridiem value is preventMeridiemChange is true. This should fix the issue and does not cause it to lose focus.

Having said that, I heard from @neilpfeiffer that the design for #706 will be to get rid of the meridiem select that is used for the desktop breakpoint and change it to be the meridiem buttons just like on the mobile viewport. If that is the plan, then the changes made in PR will no longer be needed. I wonder if we should close this PR since the meridiem input/select will be removed in favor of the meridiem buttons.

Reply:
Thanks @benbcai - you are correct that the new design adjustments to solve for #706 will be to remove the difference between the "desktop" and "mobile", and will be removing the select, instead opting for the button-group that is used in the mobile version.